src/test/javascript/spec/service/text-assessment.service.spec.ts

```diff
@@ -61,14 +79,150 @@ describe('TextAssessment Service', () => {
                             assessmentType: 'MANUAL',
                         },
                     ],
+                    blocks: [
+                        {
+                            id: '8c8d2463ec548efca05e66423bee537b6357e880',
+                            text: 'Test',
+                            startIndex: 0,
+                            endIndex: 4,
+                        },
+                    ],
                 },
             ],
         };
     });
 
-    it('should not send feedback', async () => {
-        service.trackAssessment();
-        httpMock.expectNone({ method: 'POST' });
+    it('should save an assessment', async () => {
+        let result: Result;
+        const participationId = 1;
+        result = mockResponse.submissions[0].results[0];
+        const returnedFromService = Object.assign({}, mockResponse.submissions[0].results[0]);
+
+        service
+            .save(participationId, mockResponse.submissions[0].results[0].id, mockResponse.submissions[0].results[0].feedbacks, mockResponse.submissions[0].blocks)
+            .pipe(take(1))
+            .subscribe((resp) => (expectedResult = resp.body));
+        const req = httpMock.expectOne({
+            url: `${SERVER_API_URL}api/participations/${1}/results/${6}/text-assessment`,
+            method: 'PUT',
+        });
+        req.flush(returnedFromService);
+
+        expect(expectedResult).toEqual(result);
+    });
+
+    it('should save example assessment', async () => {
+        let result: Result;
+        result = mockResponse.submissions[0].results[0];
+        const returnedFromService = Object.assign({}, mockResponse.submissions[0].results[0]);
+
+        service
+            .saveExampleAssessment(exercise.id!, mockResponse.submissions[0].id, mockResponse.submissions[0].results[0].feedbacks, mockResponse.submissions[0].blocks)
+            .pipe(take(1))
+            .subscribe((resp) => (expectedResult = resp.body));
+
+        const req = httpMock.expectOne({
+            url: `${SERVER_API_URL}api/exercises/${exercise.id}/example-submissions/${mockResponse.submissions[0].id}/example-text-assessment`,
+            method: 'PUT',
+        });
+        req.flush(returnedFromService);
+
+        expect(expectedResult).toEqual(result);
+    });
+
+    it('should submit an assessment', async () => {
+        let result: Result;
+        const participationId = 1;
+        result = mockResponse.submissions[0].results[0];
+        const returnedFromService = Object.assign({}, mockResponse.submissions[0].results[0]);
+
+        service
+            .submit(participationId, mockResponse.submissions[0].results[0].id, mockResponse.submissions[0].results[0].feedbacks, mockResponse.submissions[0].blocks)
```

Comment:
```suggestion
            .submit(participationId, result.id, result.feedbacks, mockResponse.submissions[0].blocks)
```

Reply:
Thanks! I added the exclamation mark after the property